docs/lab/06/index.md

```diff
@@ -0,0 +1,359 @@
+
+### Data transmission
+
+![SPI_protocol](images/spi_network.svg)
+
+During SPI transmission, there are 4 wires used:
+
+- **CLK** - clock - generated by the main device - subs use it to synchronously read and write to the bus
+- **MOSI** - **M**ain **O**ut **S**ub **I**n - used to carry data from the main device to *all* the subs
+- **MISO** - **M**ain **I**n **S**ub **O**ut - used to carry data from the *active* sub to the main device
+- **CS** - chip select - selects the sub to be activated, or the sub that the main wants to communicate with
+                       - one wire per sub
+                       - inactive subs will disconnect from the *MOSI* and *MISO* lines
+
+:::info
+The communication is *full-duplex*. This means that, during data transmission, both the master and the slave **must write** to the MOSI/MISO wires, no matter if that data is relevant or not.
+:::
+
+#### Transmission example
+
+1. main sets `CS` to `LOW` - the sub that the main wants to communicate with is activated - all other subs disconnect from `MOSI` and `MISO` lines
+2. main writes the first bit on the `MOSI` line, and, *simultaneously*, sub writes the first bit on the `MISO` line
+3. main starts the clock
+4. on *rising edge* 
+    - main reads data from the `MISO` line 
+    - sub reads data from the `MOSI` line
+5. on *falling edge*
+    - main writes data to the `MOSI` line
+    - sub writes data to the `MISO` line
+6. repeat steps 4 and 5 until main stops the clock
+
+:::note
+Whether steps 4 and 5 happen on rising/falling edge depends on the configuration of the SPI.
+:::
+
+![SPI_example](images/spi_network_transmission.svg)
+
+In the image above, the main first exchages data with the first sub by setting its corresponding `CS` line to low. After it finishes transmission with sub 1, it deactivates it and, afterwards, initiates communication with the second sub by setting its `CS` line to low. Notice that both main and sub read and write to the MOSI/MISO line simultaneously.
+
+### SPI modes
+
+SPI has 4 different modes which define when data is read or written. These modes can be configured through 2 registers: `CPOL` and `CPHA`.
+
+| Mode | `CPOL` | `CPHA` |
+|-|-|-|
+| 0 | 0 | 0 |
+| 1 | 0 | 1 |
+| 2 | 1 | 0 |
+| 3 | 1 | 1 |
+
+| | | |
+|-|-|-| 
+| `CPOL` | Clock polarity | defines when the data bit is read <br/> 0: *rising edge* <br/> 1: *falling edge* |
+| `CPHA` | Clock phase | defines when the data is written to the line <br/> 0: when `CS` *activates* <br/> 1: on *clock edge* (depends on `CPOL`) |
+
+### Daisy Chaining
+
+An SPI network with many sub devices can prove impractical from a hardware point of view. **Daisy chaining** is a method that solves this by linking multiple devices together with the following configuration:
```

Comment:
This does not solve the problem of many subs, it just provides another way of linking subs. Subs here usually send in MISO the previous MOSI bit.

Reply:
Removed the part about hardware impracticality and provided a clearer explanation of the way daisy chaining works. Let me know if I should add anything else here.